File: android/src/main/kotlin/io/svnk/sato_printers/models/ConnectionType.kt
package io.svnk.sato_printers.models

/**
 * Represents the type of connection to establish with a SATO printer.
 */
enum class ConnectionType {
    BLUETOOTH,
    TCP,
    USB;

    companion object {
        fun fromString(value: String): ConnectionType {
            return when (value.uppercase()) {
                "BLUETOOTH" -> BLUETOOTH
                "TCP" -> TCP
                "USB" -> USB
                else -> throw IllegalArgumentException("Unknown connection type: $value")
            }
        }
    }

    fun toMap(): String = name.lowercase()
}

File: lib/src/models/connection_type.dart
/// The type of connection to establish with a SATO printer.
enum ConnectionType {
  /// Bluetooth connection using BD address
  bluetooth,
  /// TCP/IP connection using IP address and port
  tcp,
  /// USB connection using serial number
  usb;
  /// Creates a ConnectionType from a string value.
  static ConnectionType fromString(String value) {
    switch (value.toLowerCase()) {
      case 'bluetooth':
        return ConnectionType.bluetooth;
      case 'tcp':
        return ConnectionType.tcp;
      case 'usb':
        return ConnectionType.usb;
      default:
        throw ArgumentError('Unknown connection type: $value');
    }
  }
  /// Converts to string representation.
  String toValue() => name.toLowerCase();
}
